src/WarcraftLegacies.Source/Quests/Goblin/QuestKezan.cs

```diff
@@ -31,6 +31,7 @@ public sealed class QuestKezan : QuestData
       AddObjective(new ObjectiveUpgrade(Constants.UNIT_O03N_FORTRESS_GOBLIN_T3, Constants.UNIT_O03L_GREAT_HALL_GOBLIN_T1));
       AddObjective(new ObjectiveSelfExists());
       Required = true;
+      ResearchId = Constants.UPGRADE_R09Z_QUEST_COMPLETED_OFFSHORE_INVESTMENT;
```

Comment:
It seems suspicious that you've added a ResearchId here but there's no corresponding update to RewardDescription. Does this quest enable a hero now?